JM/CF/Scripts/3_Game/CommunityFramework/Stream/CF_Stream.c

```diff
@@ -0,0 +1,183 @@
+class CF_Stream : Managed
+{
+	bool File(string path, FileMode mode)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return false;
+	}
+
+	/**
+	 * Write a byte
+	 */
+	void Write(int value)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+	}
+
+	/**
+	 * Write an ascii character
+	 */
+	void WriteChar(string value)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+	}
+
+	/**
+	 * Write a bool
+	 */
+	void WriteBool(bool value)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+	}
+
+	/**
+	 * Write an int
+	 */
+	void WriteInt(int value)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+	}
+
+	/**
+	 * Write a float
+	 */
+	void WriteFloat(float value)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+	}
+
+	/**
+	 * Write a vector
+	 */
+	void WriteVector(vector value)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+	}
+
+	/**
+	 * Write a string
+	 */
+	void WriteString(string value)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+	}
+
+	/**
+	 * Write a null terminated string
+	 */
+	void WriteCString(string value)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+	}
+
+	/**
+	 * Read a byte
+	 */
+	int Read()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return 0;
+	}
+
+	/**
+	 * Read an ascii character
+	 */
+	string ReadChar()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return "";
+	}
+
+	/**
+	 * Read a bool
+	 */
+	bool ReadBool()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return false;
+	}
+
+	/**
+	 * Read a int
+	 */
+	int ReadInt()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return 0;
+	}
+
+	/**
+	 * Read a float
+	 */
+	float ReadFloat()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return 0;
+	}
+
+	/**
+	 * Read a vector
+	 */
+	vector ReadVector()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return "0 0 0";
+	}
+
+	/**
+	 * Read a string
+	 */
+	string ReadString()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return "";
+	}
+
+	/**
+	 * Read a null terminated string
+	 */
+	string ReadCString()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return "";
+	}
+
+	bool EOF()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return true;
+	}
+
+	int Position()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return 0;
+	}
+
+	int Length()
+	{
+		/*CF.Log.*/Error("Not implemented.");
+		return 0;
+	}
+
+	void GoTo(int position)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+	}
+
+	void Seek(int offset)
+	{
+		/*CF.Log.*/Error("Not implemented.");
+	}
+
+	void SetBytes(array<int> bytes)
```

Comment:
Why is there a SetBytes method? Normally streams are only interacted with via Read and Write and the only full data operation is GetBytes?

Reply:
Syncing over the network through RPC (although inefficient).

Generally, people shouldn't use it but giving options is better than nothing